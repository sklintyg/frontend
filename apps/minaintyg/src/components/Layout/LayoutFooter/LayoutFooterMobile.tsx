import { IDSIconExternal, IDSMobileMenu, IDSMobileMenuItem } from '@frontend/ids-react-ts'
import footerLinks from '../../../data/footer-links.json'
import { LayoutFooterLink } from './LayoutFooterLink'

export function LayoutFooterMobile() {
  return (
    <>
      <IDSMobileMenu>
        {footerLinks.map(({ col, links }) =>
          links.map(({ title, url, target }) => (
            <IDSMobileMenuItem key={`${col}-${title}`}>
              <a href={url} target={target} rel="noreferrer" className="relative">
                {title}
                {target === '_blank' && (
                  <IDSIconExternal
                    className="ids-ml-2 relative -bottom-0.5"
                    inline
                    color="var(--mobile-menu-item-icon_color)"
                    height="100%"
                    width="17"
                  />
                )}
              </a>
            </IDSMobileMenuItem>
          ))
        )}
      </IDSMobileMenu>

      <span slot="sub-footer-links">
        <LayoutFooterLink href="https://www.1177.se/om-1177/1177.se/hantering-av-personuppgifter-pa-1177.se/" target="_blank">
          Behandling av personuppgifter
        </LayoutFooterLink>
      </span>
      <span slot="sub-footer-links">
        <LayoutFooterLink href="https://www.1177.se/om-1177/1177.se/hantering-av-kakor-cookies-pa-1177.se/" target="_blank">
          Hantering av kakor
        </LayoutFooterLink>
      </span>
    </>
  )
}
