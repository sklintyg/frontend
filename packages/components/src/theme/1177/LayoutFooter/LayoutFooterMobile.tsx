import { IDSIconExternal, IDSMobileMenu, IDSMobileMenuItem } from '@inera/ids-react'
import { LayoutFooterLink } from './LayoutFooterLink'
import { footerLinks } from './footerLinks'

export function LayoutFooterMobile({ hasSession }: { hasSession: boolean }) {
  return (
    <>
      <IDSMobileMenu>
        {footerLinks.map(({ col, links }) =>
          links
            .filter(({ requireSession }) => (requireSession ? hasSession : true))
            .map(({ title, url, target }) => (
              <IDSMobileMenuItem key={`${col}-${title}`}>
                <a href={url} target={target} rel="noreferrer" className="relative">
                  {title}
                  {target === '_blank' && (
                    <IDSIconExternal
                      className="ids-ml-2 relative -bottom-0.5"
                      inline
                      color="var(--mobile-menu-item-icon_color)"
                      height="100%"
                      width="17"
                    />
                  )}
                </a>
              </IDSMobileMenuItem>
            ))
        )}
      </IDSMobileMenu>

      <span slot="sub-footer-links">
        <LayoutFooterLink href="https://www.1177.se/om-1177/1177.se/hantering-av-personuppgifter-pa-1177.se/" target="_blank">
          Behandling av personuppgifter
        </LayoutFooterLink>
      </span>
      <span slot="sub-footer-links">
        <LayoutFooterLink href="https://www.1177.se/om-1177/1177.se/hantering-av-kakor-cookies-pa-1177.se/" target="_blank">
          Hantering av kakor
        </LayoutFooterLink>
      </span>
    </>
  )
}
