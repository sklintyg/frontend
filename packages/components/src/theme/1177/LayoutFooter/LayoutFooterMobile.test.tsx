import { render, screen } from '@testing-library/react'
import { LayoutFooterMobile } from './LayoutFooterMobile'

it('Should render correctly', () => {
  const { container } = render(<LayoutFooterMobile hasSession={false} />)
  expect(container).toMatchSnapshot()
  expect(screen.queryByRole('link', { name: 'Webbkarta' })).not.toBeInTheDocument()
})

it('Should have Webbkarta link when session is available', () => {
  render(<LayoutFooterMobile hasSession />)
  expect(screen.getByRole('link', { name: 'Webbkarta' })).toBeInTheDocument()
})
