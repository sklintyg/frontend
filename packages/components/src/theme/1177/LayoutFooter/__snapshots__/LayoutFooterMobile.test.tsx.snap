// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render correctly 1`] = `
<div>
  <ids-mobile-menu>
    <ids-mobile-menu-item>
      <a
        class="relative"
        href="https://www.1177.se/e-tjanster-support"
        rel="noreferrer"
        target="_blank"
      >
        Support
        <ids-icon-external
          class="ids-ml-2 relative -bottom-0.5 ids-icon"
        />
      </a>
    </ids-mobile-menu-item>
    <ids-mobile-menu-item>
      <a
        class="relative"
        href="https://www.1177.se/om-1177/1177.se/tillganglighet-pa-1177/tillganglighetsredogorelse-for-e-tjanster-pa-1177.se/"
        rel="noreferrer"
        target="_blank"
      >
        Tillgänglighet
        <ids-icon-external
          class="ids-ml-2 relative -bottom-0.5 ids-icon"
        />
      </a>
    </ids-mobile-menu-item>
    <ids-mobile-menu-item>
      <a
        class="relative"
        href="https://www.1177.se/e-tjanster"
        rel="noreferrer"
        target="_blank"
      >
        Läs mer om e-tjänster
        <ids-icon-external
          class="ids-ml-2 relative -bottom-0.5 ids-icon"
        />
      </a>
    </ids-mobile-menu-item>
  </ids-mobile-menu>
  <span
    slot="sub-footer-links"
  >
    <a
      class="text-white outline-white hover:underline"
      href="https://www.1177.se/om-1177/1177.se/hantering-av-personuppgifter-pa-1177.se/"
      rel="noreferrer"
      target="_blank"
    >
      Behandling av personuppgifter
    </a>
     
    <ids-icon-external
      class="relative -bottom-0.5 ids-icon"
    />
  </span>
  <span
    slot="sub-footer-links"
  >
    <a
      class="text-white outline-white hover:underline"
      href="https://www.1177.se/om-1177/1177.se/hantering-av-kakor-cookies-pa-1177.se/"
      rel="noreferrer"
      target="_blank"
    >
      Hantering av kakor
    </a>
     
    <ids-icon-external
      class="relative -bottom-0.5 ids-icon"
    />
  </span>
</div>
`;
